src/ecmult_multi_impl.h

```diff
@@ -159,6 +159,16 @@ static void secp256k1_ecmult_multi(secp256k1_gej *r, secp256k1_scalar *sc, secp2
              * be small. */
             secp256k1_gej_add_var(&pt[second], &pt[first], &pt[second], NULL);  /* Y -> X + Y */
             secp256k1_scalar_numsub(&sc[first], &sc[first], &sc[second]);  /* n -> n - m */
+
+            if (secp256k1_scalar_cmp_var(&sc[first], &sc[second]) < 0) {
+                break;
+            }
+
+            /* To handle pathological inputs, we use a binary ladder step here */
+            if (secp256k1_scalar_shr_int(&sc[first], 1) == 1) {
+                secp256k1_gej_add_var(r, r, &pt[first], NULL);
+            }
+            secp256k1_gej_double_nonzero(&pt[first], &pt[first], NULL);
```

Comment:
@apoelstra Just need to replace those two _double_nonzero calls with _double_var instead.